src/main/scala/Core/Wishbone.scala

```diff
@@ -0,0 +1,145 @@
+package flexpret.core
+import chisel3._
+import chisel3.util._
+import Core.FlexpretConstants._
+
+
+case class WishboneConfig(
+  numDevices: Int = 1,
+  numMasters: Int = 1
+) {
+  require(numDevices > 0)
+  require(numMasters == 1)
+
+  def nAddrBits = log2Ceil(numDevices)
```

Comment:
I think the number of address bits is a feature of the device itself, not the number of devices. The number of devices is reflected on upper bits of the address. One example would be 0xf00n---- addresses mapping all IO devices to the 0xf..... address range and using the upper bits for selecting the devices. This leaves 16 bits for each device. This is plenty for simple devices and enough for devices with (scratchpad) memory.

Reply:
You are completely right. I am conflating things here. I think the WBmaster does not really need to know where the rights are going. It only needs to know the number of address bits